opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -70,6 +70,47 @@
 ParentSpan = typing.Optional[typing.Union["Span", "SpanContext"]]
 
 
+class Link:
```

Comment:
Since these classes are so simple, why did you not keep using `namedtuple`?

Reply:
IIRC the idea is that users can extend these classes (hence the *lazy* prefix). And personally I'd prefer to keep them as very simple classes (otherwise, I'd totally be up for named tuples ;) )

Reply:
Even if those classes are very simple, I think having a proper class allows more flexibility, for instance as mentioned by @carlosalberto, when users want to extend those.

Reply:
namedtuples can be extended, they are full classes. In fact, a common pattern, that's even mentioned in the docs of namedtuple is something like:

```python
class MyDataCls(namedtuple('MyDataCls', 'foo bar')):
  def baz(self):
    return self.foo + self.bar
```

Reply:
> namedtuples can be extended

Sure, but IHMO it's still better to have this as simple classes, as it's clearer for users who want to extend it, instead of having to go read the `namedtuple` documentation ;)

(If we had this classes as private or SDK specific, I'd be also more open to have them as `namedtuple`s)

Reply:
@Oberon00 what's the benefit of using namedtuples here? If the goal is to avoid boilerplate we might want to use [the attrs library](https://www.attrs.org/en/stable/). The docs include some [reasons not to use namedtuples](https://www.attrs.org/en/stable/why.html#namedtuples) for data classes. One big reason is that subclasses of namedtuples have a MRO that looks very different from plain old classes.

Reply:
OK, these reasons sound valid enough. I retract my objections then 😃 I think since the goal of the API is to be a low level building block we should prefer a bit of boilerplate code over an additional dependency.

Reply:
Agreed, I think there's a stronger reason for using attrs in the SDK than the API, but we may not want it at all.